app/main.py

```diff
@@ -2,24 +2,83 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.protected_name = f"_{name}"
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Value type must be an integer.")
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"Value must be in {self.max_amount}"
+                             f" - {self.max_amount} range.")
```

Comment:
There is a mistake in the error message. It should include both the minimum and maximum values of the range. Currently, it repeats the maximum value twice. Consider changing the message to include both the minimum and maximum values.